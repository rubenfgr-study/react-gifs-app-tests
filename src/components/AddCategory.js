import pt from "prop-types";
import { useState } from "react";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);

    console.log("handleInputchange launch");
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <h3>{inputValue}</h3>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: pt.func.isRequired,
};
