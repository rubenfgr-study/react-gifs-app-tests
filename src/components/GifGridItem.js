import "animate.css";
import propTypes from "prop-types";
import React from "react";

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeInRight">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifGridItem.prototypes = {
  title: propTypes.string.isRequired,
  url: propTypes.string.isRequired,
};
