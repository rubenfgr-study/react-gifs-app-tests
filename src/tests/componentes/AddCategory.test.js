import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Pruebas en el component AddCategory", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe cambiar el input", () => {
    const input = wrapper.find("input");
    const value = "hello world";

    input.simulate("change", { target: { value } });

    const h3 = wrapper.find("h3");
    expect(h3.text().trim()).toBe(value);
  });

  test("no debe de mostrar la informaciÃ³n on submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).toHaveBeenCalledTimes(0);
  });

  test("debe de llamar el setCategories y limpiar la caja de texto", () => {
    wrapper.find("input").simulate("change", { target: { value: "abc" } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
