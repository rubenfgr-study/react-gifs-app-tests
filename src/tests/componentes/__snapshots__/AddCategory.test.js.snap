// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el component AddCategory Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <h3 />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
